tools/sof-dump-status.py

```diff
@@ -340,6 +341,56 @@ def _covertType(type_lst):
         print(card_info['longname'])
         exit(0)
 
+    # The kernel has changed the default firmware path when community
+    # key is used. Here we output firmware path according to kernel's
+    # match table.
+    if ret_args.get('fwpath') is True:
+        def is_community_board(board, community_boards):
+            for elem in community_boards:
+                if match_board(elem, board):
+                    return True
+            return False
+
+        # only match the key defined in community boards'
+        # match field.
+        def match_board(match, board):
+            for key in match["match"].keys():
+                if match["match"][key] != board["match"][key]:
+                    return False
+            return True
+
+        # The "community_boards" structure here is in accordance
+        # with "struct dmi_system_id community_key_platforms" in
+        # "sound/soc/sof/sof-pci-dev.c".
+        community_boards = [
+            {
+                "name": "Up Squared",
```

Comment:
why need `name`? I can understand here struct same with [community_key_platforms](https://github.com/thesofproject/linux/blob/topic/sof-dev/sound/soc/sof/sof-pci-dev.c#L40)  
But here `name` is not to use, so don't need the `name` filed

Reply:
Xiuli said we should give a name to every board, this name is for us to know which board it is.

Reply:
How about convert `name` to the comment

Reply:
I'm always having concern on the match_board(). Using this method, we will have to continue updating this file when the board list has updated. Or the board list won't be updated any more?

Reply:
@libinyang We will update board list according to sound/soc/sof/sof-pci-dev.c in the kernel. because we don't know how the kernel code change. maybe sometimes later, the kernel will add a field for matching, or add some other board, we should do the same change, too.

Reply:
@Bin-QA the kernel use "ident" , which means identity, how about this name?

Reply:
the really reason is this field is not in use, so here you struct is 
```
list[
    dict{
        # identify
        dict{
        # broad
        # vendor
        }
    }
]
```
But I think this struct is more clear & easy to query
```
dict{
    # broad 
    list [
        # vendor
    ]
}
```
python dict.get() can direct get the list

Reply:
with the first structure, we eliminate the "name" field, no need to use the second embedded dict. the second is not OK, because for some board, we need to match 2 fields, and other may be 1 fields. with a list, we need to keep field in it's place(index), but the list length is not fixed, this is error-prone.
The "name" field is not used for now, but later we may want to output which board is community board. @Bin-QA